app/Dockerfile

```
@@ -33,5 +61,73 @@ ENV HOST=0.0.0.0
 # Install application dependencies.
 # https://python-poetry.org/docs/basic-usage/#installing-dependencies
 RUN poetry install
+
 # Run the application.
 CMD ["poetry", "run", "python", "-m", "api"]
+
+#-------------------------------------------------------------------------------
+# Non-development build environment
+#-------------------------------------------------------------------------------
+FROM build as app-build
+
+# Runtime arguement for what user will the bulk of this stage run as
+ARG RUN_USER
+
+# In between ARG RUN_USER and USER ${RUN_USER}, the user is still root
+# If there is anything that needs to be ran as root, this is the spot
+
+# Change the user from root to ${RUN_USER}
+USER ${RUN_USER}
+
+# Set the application working directory.
+WORKDIR /app
+
+# Install dependencies
+# Use a project-local virtualenv to be easy to find
+ENV POETRY_VIRTUALENVS_IN_PROJECT true
```

the `ENV var val` alternative syntax is discouraged according to Docker docs and may be removed in the future, let's use the `ENV var=val` syntax (https://docs.docker.com/engine/reference/builder/#env)
```suggestion
ENV POETRY_VIRTUALENVS_IN_PROJECT=true
```

---

Good to know, in the next commit